### PR TITLE
feature: offline signing prompt

### PR DESCRIPTION
**Pull request by comountainclimber**
**What current issue(s) from Trello/Github does this address?**
This PR is the final piece of the offline signing flow... It presents a prompt to the user if they are disconnected from the internet asking if they would like to sign a transaction

**What problem does this PR solve?**
This provides an "offline" UX for users who wish to perform offline signing on an air gapped machine

**How did you solve this problem?**
I solved this problem by adding two key value pairs to state, one that checks whether the user has a connection during authentication and another that checks whether or not the user has been presented with offline prompt

**How did you make sure your solution works?**
Manual testing of many different permutations of use

**Are there any special changes in the code that we should be aware of?**

**Is there anything else we should know?**

- [ ] Unit tests written?

offline-prompt

### CODE REVIEW

File: app/actions/authActions.js [hunk 1]
```diff
@@ -35,38 +36,53 @@ type AccountType = ?{
...
   isHardwareLogin: boolean,
   isWatchOnly?: boolean,
+  hasInternetConnectivity: boolean,
 }
 
 export const ID = 'auth'
 
+export const checkForInternetConnectivity = (): Promise<boolean> =>
+  new Promise(resolve => {
+    dns.lookup('google.com', err => {
```
Comment (comountainclimber) [inline app/actions/authActions.js]: Without adding any sort of external library this seemed like the least bad option

File: app/actions/internetConnectivityPromptActions.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+export const ID = 'internet-connectivity-prompt'
+
+export const internetConnectionPromptPresented = createActions(
+  ID,
+  (bool = false) => (): {
+    internetConnectionPromptPresented: boolean,
+  } => ({
+    internetConnectionPromptPresented: bool,
+  }),
+)
```
Comment (comountainclimber) [inline app/actions/internetConnectivityPromptActions.js]: A simple boolean being added to state to determine whether or not the Dashboard should redirect to the offline prompt

File: app/actions/authActions.js (hunk 1 above)
Comment (sfreeman422) [reply app/actions/authActions.js]: `dns.resolve()` may be a better option here due to performance concerns as well as consistently using the network as mentioned in https://nodejs.org/api/dns.html#dns_implementation_considerations

File: app/actions/authActions.js [hunk 3]
```diff
@@ -35,38 +36,53 @@ type AccountType = ?{
   signingFunction?: Function,
   isHardwareLogin: boolean,
   isWatchOnly?: boolean,
+  hasInternetConnectivity: boolean,
```
Comment (comountainclimber) [inline app/actions/authActions.js]: Adding this here seemed like the best place for me as we care about performing the check for this particular flow at the point where a user signs in

**Pull request merged**